### ISSUE TITLE
Vulkan: Broken colors when baking LightmapGI

### ISSUE THREAD

**Issue by RedMser** [godotengine/godot#65586]
### Godot version

4.0.dev (ce6f284e5ff16794658db4ccd265b9c71f6b62b2)

### System information

Windows 10 64-bit, Vulkan Clustered, NVIDIA RTX 2080, Driver version 497.09

### Issue description

I was trying to debug another issue related to lightmap baking. I created a simple MRP which creates a very broken bake texture:
![image](https://user-images.githubusercontent.com/5117197/189427321-a7db5e9d-46cc-4d78-8c82-5d509d2b1d27.png)

### Steps to reproduce

- Create a simple .blend file for the scene - mark it for lightmap bake
- Add directional light node to be baked
- Create LightmapGI, set camera attributes, reduce quality to Low
- Bake and look at the results

### Minimal reproduction project

[lightmap_corruption.zip](https://github.com/godotengine/godot/files/9538122/lightmap_corruption.zip)

**Comment by Calinou** [comment]
Which graphics card model and driver version are you using?

**Comment by RedMser** [comment]
Updated the original post with specs. Since 516.94 is latest, I'll need to check after an update, but since you seem to have issues as well it might be unrelated.

**Comment by clayjohn** [comment]
> The bake works as expected in alpha 11. I don't have other alpha versions available to test, but I'll take a shot into the dark and say it's due to https://github.com/godotengine/godot/pull/63751

You have good intuition. I just tried building immediately before and after #63751 and can confirm that it introduced the issue. Fix coming in shortly.
